src/hera/workflow_template.py

```diff
@@ -0,0 +1,176 @@
+        """Adds a single task to the workflow"""
+        self.add_tasks(t)
+
+    def add_tasks(self, *ts: Task) -> None:
+        """Adds multiple tasks to the workflow"""
+        if not all(ts):
+            return
+
+        for t in ts:
+            self.spec.templates.append(t.argo_template)
+
+            if t.resources.volumes:
+                for vol in t.resources.volumes:
+                    if isinstance(vol, Volume):
+                        # dynamically provisioned volumes need associated claims on the workflow spec
+                        self.spec.volume_claim_templates.append(vol.get_claim_spec())
+                    else:
+                        self.spec.volumes.append(vol.get_volume())
+
+            self.dag_template.tasks.append(t.argo_task)
+
+    def add_head(self, t: Task, append: bool = True) -> None:
+        """Adds a task at the head of the workflow so the workflow start with the given task.
+        This sets the given task as a dependency of the starting tasks of the workflow.
+        Parameters
+        ----------
+        t: Task
+            The task to add to the head of the workflow.
+        append: bool = True
+            Whether to append the given task to the workflow.
+        """
+        if append:
+            self.add_task(t)
+
+        for template_task in self.dag_template.tasks:
+            if template_task.name != t.name:
+                if hasattr(template_task, 'dependencies'):
+                    template_task.dependencies.append(t.name)
+                else:
+                    setattr(template_task, 'dependencies', [t.name])
+
+    def add_tail(self, t: Task, append: bool = True) -> None:
+        """Adds a task as the tail of the workflow so the workflow ends with the given task.
+        This sets the given task's dependencies to all the tasks that are not listed as dependencies in the workflow.
+        Parameters
+        ----------
+        t: Task
+            The task to add to the tail of the workflow.
+        append: bool = True
+            Whether to append the given task to the workflow.
+        """
+        if append:
+            self.add_task(t)
+
+        dependencies = set()
+        task_name_to_task = dict()
+        for template_task in self.dag_template.tasks:
+            if hasattr(template_task, 'dependencies'):
+                dependencies.update(template_task.dependencies)
+            if template_task.name != t.name:
+                task_name_to_task[template_task.name] = template_task
+
+        # the tasks that are not listed as dependencies are "end tasks" i.e nothing follows after
+        # e.g if A -> B -> C then B.deps = [A] and C.deps = [B] but nothing lists C so C is "free"
+        free_tasks = set(task_name_to_task.keys()).difference(dependencies)
+        t.argo_task.dependencies = list(free_tasks)
+
+    def submit(self, namespace: Optional[str] = None) -> IoArgoprojWorkflowV1alpha1WorkflowTemplate:
```

Comment:
Since submit is deprecated, it will be removed in the next major release. What do you think about not adding it here?